epi/src/file_storage.rs

```diff
@@ -61,15 +94,8 @@ impl crate::Storage for FileStorage {
         if self.dirty {
             self.dirty = false;
 
-            let file_path = self.ron_filepath.clone();
             let kv = self.kv.clone();
-
-            std::thread::spawn(move || {
-                let file = std::fs::File::create(&file_path).unwrap();
-                let config = Default::default();
-                ron::ser::to_writer_pretty(file, &kv, config).unwrap();
-                tracing::trace!("Persisted to {:?}", file_path);
-            });
+            self.sender.send(Message::DoFlush(kv)).unwrap();
```

Comment:
The problem with this is that there is no backpressure in `std::sync::mpsc::channel`, because it is an infinite buffer. This means that if serialization is slow, we will just create an ever-growing queue of `Message::DoFlush` until we run out of memory.

I think the simpler solution is to just store a join handle from `std::thread::spawn` and join any old thread before spawning a new one. Considering the default auto-save policy is 30s, the overhead of spawning a thread is negligible in comparison.